python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
thanks @jerryshao for kindly review

1. SparkSubmit has logic of forbidding running shell with cluster mode. So I think it's enough to ignore "deploy mode" here.
2. Furthermore, I guess test "master" here and in SparkILoop.scala may be redundant, because:
  in SparkContext.scala, it has: 
`if (master == "yarn" && deployMode == "client") System.setProperty("SPARK_YARN_MODE", "true")`

  in SparkSubmit.scala, it has: 
`    if (clusterManager == YARN) {
      if (args.isPython) {
        sysProps.put("spark.yarn.isPython", "true")
      }  
`

  Another possible proof is that after SPARK-13220, test 'yarn-client' will always get false, code under this **_if_** block would never be executed, but no one report related error.